### DISCUSSION TITLE
Integrated terminal

### DISCUSSION THREAD

**Discussion by apiening** [LunarVim/LunarVim#3794] [Q&A] [answered]
Just to provide some context, I'm in the process of switching from https://github.com/ChristianChiarulli/nvim to `LunarVim`.

Since the shortcut overview that pops up after a short delay when the `<leader>` key is pressed did not show **Terminal** related options, I wanted to mimic the behavior of the config from [Chris@machine](https://www.chrisatmachine.com/) where a submenu reachable by `<leader> <t>` shows options for multiple terminals and options for `Float`, `Horizontal`, `Vertical` and others.

I noticed that `ToggleTerm` which seems to be responsible for the floating terminal, is already installed. I added the following config to my `config.lua`:
```lua
lvim.builtin.which_key.mappings["t"] = {
  name = "+Terminal",
  f = { "<cmd>ToggleTerm<cr>", "Floating terminal" },
}
```
This gives me the option to start a floating terminal with `<leader> <t> <f>` and the shortcut is also displayed in the shortcut overview popup.

But I'm still missing the other terminal options (horizontal, vertical, multiple terminals etc.) and I wonder how they are implemented. It feels like I'm missing something.

Any hints are greatly appreciated.

**Comment by brneor** [comment, marked as answer]
You can try something like

```lua
lvim.builtin.which_key.mappings["t"] = {
  name = "+Terminal",
  f = { "<cmd>ToggleTerm<cr>", "Floating terminal" },
  v = { "<cmd>2ToggleTerm size=30 direction=vertical<cr>", "Split vertical" },
  h = { "<cmd>2ToggleTerm size=30 direction=horizontal<cr>", "Split horizontal" },
}
```

Got it from [here](https://github.com/ChristianChiarulli/nvim/blob/master/lua/user/toggleterm.lua).

**Reply by apiening** [reply]
Thank you @brneor, this is working fine for me!

**Comment by xbz-24** [comment]
tysm guys this works great but how do I quickly exit the floating terminal?

**Reply by apiening** [reply]
I just do `<control> <d>`.

**Reply by mcp292** [reply]
You can also type "exit".

**Comment by InAnYan** [comment]
But why is this not a default...

Integrated terminal is a crucial feature...
